http/src/test/java/com/redhat/lightblue/client/http/testing/doubles/FakeLightblueRequest.java

```diff
@@ -0,0 +1,31 @@
+package com.redhat.lightblue.client.http.testing.doubles;
+
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.request.LightblueRequest;
+
+public class FakeLightblueRequest implements LightblueRequest {
```

Comment:
The issues I see:
* As you said, too many clients: lightblue client, specific http client implementation wrapper and specific http client implementation. First 2 layers are thin wrappers and carry little substance.
* Implementing header handling for each http client implementation feels redundant, though perhaps it's unavoidable if multiple http client implementations are to be supported.
* Actually, I don't see much benefit of being able to switch http client implementations. I would decide on one and stick to it (it could still be changed if need be, just not so conveniently).

So far I implemented compression functionality as you suggested. This is not a big deal, I just have an impression that you are over-generalizing things a bit. If someone said they want to be able to switch http clients, then I'd rest my case.